src/smart-components/role/edit-resource-definitions-modal.js

```diff
@@ -2,32 +2,45 @@ import React, { useEffect, useReducer } from 'react';
 import PropTypes from 'prop-types';
 import componentTypes from '@data-driven-forms/react-form-renderer/component-types';
 import FormRenderer from '../common/form-renderer';
+import flatten from 'lodash/flattenDeep';
+import { FormattedMessage, useIntl } from 'react-intl';
 import { useParams } from 'react-router-dom';
 import { useDispatch, useSelector, shallowEqual } from 'react-redux';
 import { updateRole, fetchRole } from '../../redux/actions/role-actions';
-import { getResource, getResourceDefinitions } from '../../redux/actions/cost-management-actions';
+import { fetchResource, fetchResourceDefinitions } from '../../redux/actions/cost-management-actions';
+import { fetchInventoryGroups } from '../../redux/actions/inventory-actions';
 import componentMapper from '@data-driven-forms/pf4-component-mapper/component-mapper';
 import WarningModal from '@patternfly/react-component-groups/dist/dynamic/WarningModal';
 import { Spinner, Modal, ModalVariant, Bullseye } from '@patternfly/react-core';
 import useAppNavigate from '../../hooks/useAppNavigate';
 import ResourceDefinitionsFormTemplate from './ResourceDefinitionsFormTemplate';
-import flatten from 'lodash/flattenDeep';
-import { useIntl } from 'react-intl';
+import { HOSTS_TYPE, INVENTORY_PREFIX } from '../../utilities/constants';
 import messages from '../../Messages';
 import './role-permissions.scss';
 
-const createOptions = (resources) =>
-  Object.entries(resources).reduce(
-    (acc, [key, value]) => [
-      ...acc,
-      ...value.map((r) => ({
-        value: r.value,
-        path: key,
-        label: r.value,
-      })),
-    ],
-    []
-  );
+const createOptions = (resources, isInventory, isHosts) =>
```

Comment:
We may want to adjust this later down the line as notifications will be added to this functionality.